controllers/check_client_compatibility.go

```diff
@@ -102,5 +94,47 @@ func (c checkClientCompatibility) reconcile(_ context.Context, r *FoundationDBCl
 		return &requeue{message: message, delay: 1 * time.Minute}
 	}
 
+	if compatibleClientsCount != status.Cluster.Clients.Count {
+		message := fmt.Sprintf("expected to find %d compatible clients for version %s, but only got %d",
+			status.Cluster.Clients.Count,
+			cluster.Spec.Version,
+			compatibleClientsCount)
+		r.Recorder.Event(cluster, corev1.EventTypeNormal, "UnsupportedClient", message)
+		logger.Info("Deferring reconciliation due to missing compatible clients", "message", message)
+		return &requeue{message: message, delay: 1 * time.Minute}
+	}
+
 	return nil
 }
+
+// getUnsupportedClients returns the list of unsupported clients for this protocol version and desired version. It also returns the number of clients
+// connected with a protocol compatible version.
+func getUnsupportedClientsAndCheckIfCompatibleClientsAreIncluded(supportedVersions []fdbv1beta2.FoundationDBStatusSupportedVersion, protocolVersion string, desiredVersion fdbv1beta2.Version) ([]string, int, error) {
+	var unsupportedClients []string
+	var compatibleClientsCount int
+
+	for _, versionInfo := range supportedVersions {
+		if versionInfo.ProtocolVersion == "Unknown" {
+			continue
+		}
+
+		parsedVersion, err := fdbv1beta2.ParseFdbVersion(versionInfo.ClientVersion)
+		if err != nil {
+			return nil, -1, err
+		}
+
+		if versionInfo.ProtocolVersion == protocolVersion {
+			compatibleClientsCount += versionInfo.Count
```

Comment:
What's the behavior here if a client has two multi-version libraries with the same protocol version? Do they get double-counted, or do they only get counted under one of the two libraries?